pandas/core/dtypes/common.py

```diff
@@ -599,11 +599,14 @@ def is_dtype_equal(source, target) -> bool:
         if not isinstance(source, str):
             # GH#38516 ensure we get the same behavior from
             #  is_dtype_equal(CDT, "category") and CDT == "category"
+            # GH-44276 ensures behaviour is consistent
+            # is_dtype_equal (dtype, "string[pyarrow]")
             try:
                 src = get_dtype(source)
-                if isinstance(src, ExtensionDtype):
+                if isinstance(src, (ExtensionDtype, np.dtype)):
```

Comment:
is this change still needed?

Reply:
It is needed because dtypes such as ```int64``` ```float64``` and other dtypes actually are ```np.dtype```, so if I am not adding this check, the test is failing.

Reply:
> It is needed because dtypes such as int64 float64 and other dtypes actually are np.dtype, so if I am not adding this check, the test is failing.

It is not needed if we catch ImportError more generally, I think; I pushed some small commits to verify this.